contracts/test/ContractAddress.js

```diff
@@ -0,0 +1,30 @@
+const { assert } = require("chai");
+const { ethers, web3 } = require("hardhat");
+
+describe("ContractAddress", function () {
+    it("Contract address should be null when not deploy contract", async function () {
+        const Storage = await ethers.getContractFactory("Calc");
+        const contract = await Storage.deploy();
+        await contract.deployed();
+
+        //Set data
+        const tx = await contract.store(256);
+        await tx.wait();
```

Comment:
```suggestion
        await tx.wait(2);
```

this should fix the instant-finality problem and we can remove the rest while loop.

Reply:
Updated